src/components/Dropzone/Dropzone.vue

```diff
@@ -0,0 +1,345 @@
+      </label>
+
+      <button
+        v-if="successStep"
+        type="reset"
+        aria-label="Remove file"
+        class="mt-6 underline text-gray-500 !text-sm"
+        @click.stop="removeFile"
+      >
+        {{ textContent.removeFileButtonText }}
+      </button>
+
+      <ProgressBar
+        v-if="uploadingStep"
+        :percentage="progress"
+        class="mt-6"
+      />
+    </form>
+    <div class="mt-2 flex items-center justify-between">
+      <div
+        v-if="showTypeAndMaxSize"
+        :class="['text-sm', errorStep ? 'text-error' : 'text-gray-300']"
+      >
+        <span v-if="acceptType && acceptType !== '/*'">
+          <span v-if="fileTypesArray.length === 1">
+            {{ textContent.acceptedFormatIs }} <strong>{{ fileTypesString }}</strong>.
+          </span>
+          <span v-if="fileTypesArray.length > 1">
+            {{ textContent.acceptedFormatsAre }} <strong>{{ fileTypesString }}</strong>.
+          </span>
+        </span>
+        <span v-if="maxSizeInBytes">
+          {{ textContent.maxFileSizeIs }} <strong>{{ formatBytes(maxSizeInBytes) }}</strong>.
+        </span>
+      </div>
+      <LobLink
+        v-if="sampleLinkUrl"
+        :to="sampleLinkUrl"
+        target="_blank"
+        class="!text-sm"
+      >
+        {{ textContent.downloadSampleFile }}
+      </LobLink>
+    </div>
+  </div>
+</template>
+
+<script>
+import { formatBytes } from '@/utils/formatBytes';
+import { LobLink, ProgressBar, Upload } from '@/components';
+
+export default {
+  name: 'Dropzone',
+  components: { LobLink, ProgressBar, Upload },
+  props: {
+    textContent: { type: Object, required: true,
+      validator: function (value) {
+        const requiredKeys = ['dragAndDropHere', 'yourFile', 'or', 'uploadFileButtonText',
+          'acceptedFormatIs', 'acceptedFormatsAre', 'maxFileSizeIs', 'downloadSampleFile',
+          'uploading', 'mightTakeAMinute', 'looksGreat', 'removeFileButtonText',
+          'couldNotUpload', 'canOnlySelectOneFile', 'fileIsTooLarge', 'fileTypeNotValid',
+          'defaultErrorText', 'errorMessage', 'successMessage'];
+        for (const key of requiredKeys) {
+          return (Object.keys(value).includes(key));
+        }
+      }
+    },
+    uploadId: { type: String, required: true },
+    acceptType: { type: String, default: '/*' },
+    maxSizeInBytes: { type: [Number, null], default: null },
+    showTypeAndMaxSize: { type: Boolean, default: true },
+    sampleLinkUrl: { type: String, default: null },
+    progress: { type: [Number, null], default: null },
+    fileOb: {  type: Object, default: null },
+    status: { type: [String, null], default: null,
+      validator: function (value) {
+        return [null, 'error', 'success'].includes(value);
+      }
+    }
+  },
+  emits: ['select', 'remove'],
+  data: function () {
+    return {
+      formatBytes,
+      selectedFile: this.fileOb,
+      currentStep: this.status !== null ? this.status :  'default',
```

Comment:
Instead here we could do:
```suggestion
      currentStep: this.status ?? 'default',
```

(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)